### ISSUE TITLE
Finalize scope and strategy for metadata enrichment

### ISSUE THREAD

**Issue by ssarrafan** [microbiomedata/nmdc-metadata#330]
Chris and/or team to review milestone 1.4 and finalize the scope and strategy around metadata enrichment.  Additional tasks can be created based on the strategy and scope.

**Comment by cmungall** [comment]
Breaking 1.3 down and providing a unifying framework for integrating the different pieces. These are not all interdependent but could be done sequentially, although the base framework should be in place first.

 - [ ] Write enrichment framework python library. The core framework will not perform any enrichment, it will provide the backbone from which individual routines can be implemented. @wdduncan
 - [ ] add NER routine for suggesting biome terms based on package + sample description, limited to soil/sediment package for first pass @turbomam 
 - [ ] ditto for material and feature
 - [ ] ditto for fao_soil_class or other mixs controlled field of relevance to NMDC
 - [ ] plug in @StantonMartin's code for ORNL Identify elevation layer (we can choose another quantitive layer, this seems easy one)
 - [ ] plug in @StantonMartin's code for ORNL soil type layer (we can choose another qualitative layer, this seems easy one)
 - [ ] Wrap python library in flask or fastapi, exposing an openAPI endpoint (do we have NMDC API standards?)
 - [ ] provide demonstrator of how this can be integrated e.g. with a google sheet or gold database

Differences from 1.3 workplan: the underlying lib we use for NER is no longer SciGraph, Mark is using a mixture of services and OGER

Deferred from workplan/TBD: "metabolite sample descriptions" 

The python framework will take as input a non-necessarily-structured NMDC sample object as a simple dict, e.g.

```yaml
      id: "gold:Gb0108335"
      name: "Thawing permafrost microbial communities from the Arctic, studying carbon transformations - Permafrost 712P3D"
      lat_lon: 
        has_raw_value: "68.3534 19.0472"
        latitude: 68.3534
        longitude: 19.0472
        type: "nmdc:GeolocationValue"
      collection_date: 
        has_raw_value: "2012-07-20"
      ecosystem: 
        has_raw_value: "Environmental"
      ecosystem_category: 
        has_raw_value: "Terrestrial"
      ecosystem_type: 
        has_raw_value: "Soil"
      ecosystem_subtype: 
        has_raw_value: "Wetlands"
      specific_ecosystem: 
        has_raw_value: "Permafrost"
      depth: 
        has_raw_value: "0.0"
      ncbi_taxonomy_name: 
        has_raw_value: "permafrost metagenome"
      sample_collection_site: 
        has_raw_value: "Palsa"
```

and then suggest missing values, enhancements, potential problems. We will have a simple QC schema for this. Payload could be something like:

```yaml
annotation_sufficiency: 0.43
description: <narrative text summarizing sufficiency metrics>
suggestions:
  - type: location_inference
     sample_ids: [gold:Gb0108335]
     field: geo_loc_name
     value: Sweden: Kiruna
     confidence: HIGH
     was_generated_by:
       <prov information about identify/wikidata API call to fetch location name from geolocation>
  - type: envo_inference
     sample_ids: [gold:Gb0108335]
     field: env_broad_scale
     value: 
       term:
          id: ENVO:00000446
          name: terrestrial biome
     confidence: MEDIUM
     was_generated_by:
       <prov information about NER API call to infer ENVO biome from text>
  - ...
problems:
  - type: MISSING_UNIT
     sample_ids: [gold:Gb0108335]
     field: depth
     value: "0.0"
     inferred_value:
        has_raw_value: "0.0 m"
        unit: m
        value: 0.0
```

**Comment by dwinston** [comment]
> Wrap python library in flask or fastapi, exposing an openAPI endpoint (do we have NMDC API standards?)

@cmungall what do you mean by "NMDC API standards"?

I am developing (with Bin, Shane, and Yuri) an OpenAPI-compliant API (via FastAPI) for workflow automation / data management at [microbiomedata/nmdc-runtime](https://github.com/microbiomedata/nmdc-runtime/tree/03ab944e840acd19522218e5c6ee45b1ae6587be/nmdc_runtime/api). We can extend this API to include a metadata enrichment endpoint.

**Comment by ssarrafan** [comment]
Thanks @cmungall. I'l move this to June.

**Comment by ssarrafan** [comment]
Discussed with @emileyfadrosh and @turbomam and assigning to Mark to do some pieces of this. Will follow up with Chris on outstanding items that Mark won't be doing.  Adding this to the July sprint.
